README.md

```diff
@@ -492,6 +495,52 @@ status:
     status: 'True'
 ```
 
+## Custom Binding Paths
+
+There are scenarios where the application resource is not PodSpec-able with custom binding paths for env, volumes, and volumeMounts.  This extension allows a `ServiceBinding` resource to specify custom binding paths for the application resource.  The custom binding paths **MUST** be one or more of env, volumes, and volumeMounts.  In the absence of any of these items, that particular projection **MUST NOT** be performed.  The `ServiceBinding` author **MUST** set the `service.binding/path` annotation to `Custom`.  The binding path value **MUST** be a [JSONPath][jp] location in the application resource.  The binding paths **MUST** be specified in a ConfigMap, and `.spec.application.bindingPaths` in `ServiceBinding` resource **MUST** refer to it.
+
+[jp]: https://kubernetes.io/docs/reference/kubectl/jsonpath/
+
+### Custom Binding Paths Example ConfigMap
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-binding-paths
+data:
+  env: "{.spec.env}"
+  volumes: "{.spec.volumes}"
+  volumeMounts: "{.spec.volumeMounts}"
+```
+
+### Custom Binding Paths Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ServiceBinding
+metadata:
+  name: account-service
+  annotations:
+    service.binding/path: "Custom"
+spec:
+  application:
+    apiVersion:    apps.example.org/v1
+    kind:          ExampleResource
+    name:          online-banking
+    bindingPaths:  example-binding-paths
```

Comment:
Do we want every binding creator to need to specify this value?

How likely are these values to vary between different resources of the same kind? Can we key the custom mapping based on the apiVersion and kind?

What should happen if this ConfigMap doesn't exist, or doesn't contain the keys we expect, or the JSONPaths in those keys are invalid?

Reply:
I have changed the approach in the modified text. I think it has addressed all these questions.